src/System.CommandLine/ChildList.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections;
+using System.Collections.Generic;
+
+namespace System.CommandLine
+{
+    /// <summary>
+    /// a wrapper of List<typeparamref name="T"/> that sets parent for every added element
+    /// </summary>
+    internal sealed class ChildList<T> : IList<T> where T : Symbol
```

Comment:
Did you consider this?
```suggestion
    internal sealed class ChildList : IList<Symbol>
```

Reply:
It would be impossible to enforce the right types without perf overhead. Example:

```cs
Option<bool> option = new ("--some);
RootCommand root = new ();
// in the line below Arguments would be a list of Symbols, which would allow for adding not just arguments but also options and commands
root.Arguments.Add(option); // no compilation error, at runtime we would later find out that it's not an Argument
```